pkg/policy/ebpf.go

```diff
@@ -0,0 +1,726 @@
+			binary.LittleEndian.PutUint32(binBytes, k.MntNS)
+			copy(binBytes[4:], k.Path)
+		}
+		keyPointer := unsafe.Pointer(&binBytes[0])
+
+		eqVal := make([]byte, equalityValueSize)
+		valuePointer := unsafe.Pointer(&eqVal[0])
+
+		binary.LittleEndian.PutUint64(eqVal[0:8], v.equalInPolicies)
+		binary.LittleEndian.PutUint64(eqVal[8:16], v.equalitySetInPolicies)
+
+		bpfMap, ok := ps.bpfInnerMaps[innerMapName]
+		if !ok {
+			return errfmt.Errorf("bpf map not found: %s", innerMapName)
+		}
+		if err := bpfMap.Update(keyPointer, valuePointer); err != nil {
+			return errfmt.WrapError(err)
+		}
+	}
+
+	return nil
+}
+
+type procInfo struct {
+	newProc        bool
+	followPolicies uint64
+	mntNS          uint32
+	binaryBytes    [maxBpfBinPathSize]byte
+	binNoMnt       uint32
+}
+
+// TODO: Should ProcInfoMap be cleared when a Policies new version is created?
+// Or should it be versioned too?
+func populateProcInfoMap(bpfModule *bpf.Module, binEqualities map[filters.NSBinary]equality) error {
+	procInfoMap, err := bpfModule.GetMap(ProcInfoMap)
+	if err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	binsProcs, err := proc.GetAllBinaryProcs()
+	if err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	// populate proc info map with binaries to track
+	for bin := range binEqualities {
+		procs := binsProcs[bin.Path]
+		for _, p := range procs {
+			binBytes := make([]byte, maxBpfBinPathSize)
+			copy(binBytes, bin.Path)
+			binBytesCopy := (*[maxBpfBinPathSize]byte)(binBytes)
+			// TODO: Default values for newProc and followPolicies are 0 are safe only in
+			// init phase. As Policies are updated at runtime, this is not true anymore.
+			procInfo := procInfo{
+				newProc:        false,
+				followPolicies: 0,
+				mntNS:          bin.MntNS,
+				binaryBytes:    *binBytesCopy,
+				binNoMnt:       0, // always 0, see bin_no_mnt in tracee.bpf.c
+			}
+			if err := procInfoMap.Update(unsafe.Pointer(&p), unsafe.Pointer(&procInfo)); err != nil {
+				return errfmt.WrapError(err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (ps *Policies) UpdateBPF(
+	bpfModule *bpf.Module,
+	cts *containers.Containers,
+	eventsState map[events.ID]events.EventState,
+	eventsParams map[events.ID][]bufferdecoder.ArgType,
+	updateProcTree bool,
+) error {
+	ps.rwmu.Lock()
+	defer ps.rwmu.Unlock()
+
+	// Create new events map version
+	if err := ps.createNewEventsMapVersion(bpfModule, eventsState, eventsParams); err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	fEqs := &filtersEqualities{
+		uidEqualities:      make(map[uint64]equality),
+		pidEqualities:      make(map[uint64]equality),
+		mntNSEqualities:    make(map[uint64]equality),
+		pidNSEqualities:    make(map[uint64]equality),
+		cgroupIdEqualities: make(map[uint64]equality),
+		utsEqualities:      make(map[string]equality),
+		commEqualities:     make(map[string]equality),
+		procTreeEqualities: make(map[uint64]equality),
+		binaryEqualities:   make(map[filters.NSBinary]equality),
+	}
```

Comment:
You could document all this much better. What this function does, when is it called, why. Also, the maps. Index is UID, then PID then MNTNS, etc. You could have a description describing equality (meaning in what policy that is enabled, and in which one is is a "!=" value). Etc. This code is way too obscure for an open source project (not because of you). It is very hard to any 3rd party contributor to understand what goes on in this part.